transport/mapping_serializer.go

```diff
@@ -55,13 +57,16 @@ func (m *MappingSerializer[T]) Deserialize(streamID stream.ID, name stream.Event
 	}
 
 	event := mapper.StorageEvent()
+	newEvent := reflect.New(reflect.TypeOf(event)).Interface()
 
-	err = m.marshaler.Unmarshal(streamID, payload, event)
+	err = m.marshaler.Unmarshal(streamID, payload, &newEvent)
```

Comment:
I discovered it today. If you want to instantiate a T type, try `new(T)` you will have a pointer to it right away :0

Reply:
I guess you could avoid some of that reflection wizardry.

Reply:
In this case, the `T` is the `Stream`, not the event.